optuna/trial.py

```diff
@@ -23,6 +24,244 @@
+            raise ValueError("`datetime_start` is supposed to be set.")
+
+        if self.state.is_finished():
+            if self.datetime_complete is None:
+                raise ValueError("`datetime_complete` is supposed to be set for a finished trial.")
+        else:
+            if self.datetime_complete is not None:
+                raise ValueError(
+                    "`datetime_complete` is supposed to be None for an unfinished trial."
+                )
+
+        if self.state == TrialState.COMPLETE and self.value is None:
+            raise ValueError("`value` is supposed to be set for a complete trial.")
+
+        if set(self.params.keys()) != set(self.distributions.keys()):
+            raise ValueError(
+                "Inconsistent parameters {} and distributions {}.".format(
+                    set(self.params.keys()), set(self.distributions.keys())
+                )
+            )
+
+        for param_name, param_value in self.params.items():
+            distribution = self.distributions[param_name]
+
+            param_value_in_internal_repr = distribution.to_internal_repr(param_value)
+            if not distribution._contains(param_value_in_internal_repr):
+                raise ValueError(
+                    "The value {} of parameter '{}' isn't contained in the distribution "
+                    "{}.".format(param_value, param_name, distribution)
+                )
+
+    @property
+    def distributions(self):
+        # type: () -> Dict[str, BaseDistribution]
+        """Return the distributions for this trial.
+
+        Returns:
+            The distributions.
+        """
+
+        return self._distributions
+
+    @distributions.setter
+    def distributions(self, value):
+        # type: (Dict[str, BaseDistribution]) -> None
+        """Set the distributions for this trial.
+
+        Args:
+            value: The distributions.
+        """
+
+        self._distributions = value
+
+    @property
+    def trial_id(self):
+        # type: () -> int
+        """Return the trial ID.
+
+        .. deprecated:: 0.19.0
+            The direct use of this attribute is deprecated and it is recommended that you use
+            :attr:`~optuna.trial.FrozenTrial.number` instead.
+
+        Returns:
+            The trial ID.
+        """
+
+        warnings.warn(
+            "The use of `FrozenTrial.trial_id` is deprecated. "
+            "Please use `FrozenTrial.number` instead.",
+            DeprecationWarning,
+        )
+
+        logger = logging.get_logger(__name__)
+        logger.warning(
+            "The use of `FrozenTrial.trial_id` is deprecated. "
+            "Please use `FrozenTrial.number` instead."
+        )
+
+        return self._trial_id
+
+    @property
+    def last_step(self):
+        # type: () -> Optional[int]
+
+        if len(self.intermediate_values) == 0:
+            return None
+        else:
+            return max(self.intermediate_values.keys())
+
+
```

Comment:
Ops, I believe changes in this file are unintended.

Reply:
Thank you for your comment. Yes, you are right. I will remove them.